design/one-pager-package-format-v2.md

```diff
@@ -0,0 +1,248 @@
+  the provider, and provides metadata for systems that build on Crossplane -
+  i.e. package registries.
+* The `packages.crossplane.io` `Provider` is submitted to the Crossplane API
+  server in order to declare that a provider should be installed and run. The
+  Crossplane package manager runs a provider by unpacking its package and
+  extracting its `meta.packages.crossplane.io` `Provider` configuration.
+
+Note that the two have different audiences; the `meta.packages.crossplane.io`
+file is authored by the provider maintainer, while the `packages.crossplane.io`
+custom resource is authored by the platform operator.
+
+Contemporary packages contain an `install.yaml` file that allows the package
+author to specify how the package's controller (e.g. the provider) should be
+invoked. This document proposes support for `install.yaml` be removed; packages
+will now only install a specific kind of controller - a Crossplane provider. It
+is possible that both the maintainer of the provider and the platform operator
+will need to influence how the provider is deployed; e.g. by specifying env
+vars, health checks, replica counts, or node selectors. Currently all providers
+use a minimal `install.yaml`; they do not make use of their ability to influence
+how the provider is run. This document recommends that the above `Provider`
+types be extended when and as appropriate to configure these concerns.
+
+### Packaging a Configuration
+
+A packaged Crossplane configuration must be an OCI image consisting of:
+
+1. A `crossplane.yaml` file containing a `kind: Configuration` at its root.
+1. One or more `CompositeResourceDefinition` and/or `Composition` resources.
+1. Optional additional metadata that is opaque to Crossplane in a `.crossplane`
+   directory at the root of the package.
+
+Below is an example `Configuration`, at `v1alpha`:
+
+```yaml
+# Required. Must be as below.
+apiVersion: meta.packages.crossplane.io/v1alpha1
+# Required. Must be as below.
+kind: Configuration
+# Required. Note that Crossplane is aware only of the name and annotations
+# metadata fields. Other fields (e.g. labels) will be preserved but opaque.
+metadata:
+  # Required. Must comply with Kubernetes API conventions.
+  name: configuration-example
+  # Optional. Must comply with Kubernetes API conventions. Annotations are
+  # opaque to Crossplane, which will replicate them to the annotations of a
+  # PackageRevision when this package is unpacked, but otherwise ignore them.
+  # Systems such as package registries may extend this specification to require
+  # or encourage specific annotations.
+  annotations:
+    company: Upbound
+    maintainer: Nic Cope <negz@upbound.io>
+    keywords: cloud-native, kubernetes, example
+    source: github.com/crossplane-contrib/config-example
+    license: Apache-2.0
+    description: |
+      The Example configuration adds example resources to Crossplane.
+    provider: example
+# Required.
+spec:
+  dependsOn:
```

Comment:
```suggestion
spec:
  # Optional. Specifies a Crossplane version that the package is compatible with.
  crossplane: v0.13.0
  # Optional. Used by Crossplane to ensure any dependencies of a configuration
  # installed and running before the configuration is installed.
  dependsOn:
```

Reply:
Just a note that in some of my sketches I interpreted what is actually `spec.crossplane` as having been an entry under `spec.dependsOn`. I prefer it as you have here (`spec.crossplane`) because it has a different format than the `dependsOn` entries.

```yaml
dependsOn:
- type: Configuration
  image: exampleorg/myconfig
  version: "v0.1.0"
- type: Provider
  image: crossplane/provider-example
  version: "v0.1.0"
- type: Crossplane
  version: "v0.13.0"
```

I considered something like the above, which I think would also work if we wanted something a little more verbose. I don't love `image` in the above suggestion though, nor `package` or `name` as alternatives.

All in all, I think what you're suggesting here is the way to go.